yarn/src/main/scala/org/apache/spark/deploy/yarn/security/AMDelegationTokenRenewer.scala

```diff
@@ -69,6 +71,9 @@ private[yarn] class AMDelegationTokenRenewer(
   private val freshHadoopConf =
     hadoopUtil.getConfBypassingFSCache(hadoopConf, new Path(credentialsFile).toUri.getScheme)
 
+  @volatile private var timeOfNextRenewal =
+    sparkConf.get(CREDENTIALS_RENEWAL_TIME).getOrElse(Long.MaxValue)
```

Comment:
One possible scenario from current code is that if principal and keytab is configured, but all credential are non-renewable or disabled. In such case `CREDENTIALS_RENEWAL_TIME` may be empty but `AMDelegationTokenRenewer` will still be started because it only relies on principal configured or not.